### PR TITLE
Support for Re-Ranker via Text Embedding Interface

### PR DESCRIPTION
**Pull request by Rohith-2**
# Description

Text Embedding Inference by Huggingface supports hosting re-ranker models, which at the moment are not supported by llama-index post processing module. 

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [X] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [X] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Added new unit/integration tests
- [X] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/README.md
Comment (logan-markewich) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/README.md:1]: Can we put a sample install command and some example of usage? This readme will show up on https://llamahub.ai

File: llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+python_version = "3.8"
+
+[tool.poetry]
+authors = ["Rohith Ramakrishnan <rrohith2001@gmail.com>"]
+description = "llama-index postprocessor text embedding inference rerank integration"
+exclude = ["**/BUILD"]
+license = "MIT"
+name = "llama-index-postprocessor-tei-rerank"
+readme = "README.md"
+version = "0.1.6"
```
Comment (logan-markewich) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/pyproject.toml:30]: probably should start at v0.1.0?

Comment (Rohith-2) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/pyproject.toml]: yup, my bad! forgot to change this.

File: llama-index-core/llama_index/core/postprocessor/__init__.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
     PIINodePostprocessor,
 )
 from llama_index.core.postprocessor.sbert_rerank import SentenceTransformerRerank
+from llama_index.core.postprocessor.tei_rerank import TextEmbeddingInference
```
[the remark this answers is not in the document]
Comment (Rohith-2) [reply llama-index-core/llama_index/core/postprocessor/__init__.py]: Sure, made the changes.

File: llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/README.md
Comment (Rohith-2) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/README.md:1]: Can do, will make the necessary changes and push it soon.

Comment (Rohith-2) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-tei-rerank/README.md:1]: Pushed the changes, hope that covers it 😃

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**